tests/python_package_test/test_dask.py

```diff
@@ -207,6 +235,15 @@ def test_classifier(output, centers, client, listen_port):
     assert_eq(p1_local, p2)
     assert_eq(y, p1_local)
 
+    # be sure LightGBM actually used at least one categorical column
+    if output == 'dataframe-with-categorical':
+        cat_cols = [
+            col for col in dX.columns
+            if dX.dtypes[col].name == 'category'
+        ]
+        tree_df = dask_classifier.booster_.trees_to_dataframe()
+        assert tree_df['split_feature'].isin(cat_cols).sum() > 0
```

Comment:
I'm not sure this assert is enough to say that at least one of `cat_cols` was categorical. I believe you should check for comparison sign as well to ensure these features were treated as categorical.

Reply:
My goal was not to check that the columns are treated as categorical features by LightGBM. My goal was to ensure that having pandas "category" columns in your training data does not break LightGBM's Dask estimators (as it currently does on `master`, documented in #3861).

As far as I can tell, "category" columns will only be  automatically treated as categorical features if you set  `categorical_feature = 'auto'`: https://github.com/microsoft/LightGBM/blob/8ef874bd4804b54b48ef4d796c8110be671a6950/python-package/lightgbm/basic.py#L513-L514, [which is not the default](https://lightgbm.readthedocs.io/en/latest/Parameters.html#categorical_feature), and of course you can force it to happen by setting `categorical_featue` to the column names / indices in the parameters.

The tests like `assert_eq(p1_local, p2)` already test that the Dask interface is producing a model that is similar to the one produced by the scikit-learn interface.

I can add `categorical_feature = [col for col in dX.columns if col.startswith('cat_')]` and test, that's fine. I will try to do that tonight. But it wasn't the intention of this PR or these tests.

Reply:
I've added these new checks in https://github.com/microsoft/LightGBM/pull/3908/commits/07c9dcaeed7150be32c0d97fd26e7d222a4dba9d

Reply:
Sorry for I didn't get the purpose of this PR!